src/toil/lib/docker.py

```diff
@@ -1,280 +1,340 @@
-           If list of lists: list[list[str]], then treat as successive commands chained with pipe.
-    :param str workDir: Directory to mount into the container via `-v`. Destination convention is /data
-    :param list[str] dockerParameters: Parameters to pass to Docker. Default parameters are `--rm`,
-            `--log-driver none`, and the mountpoint `-v work_dir:/data` where /data is the destination convention.
-             These defaults are removed if docker_parmaters is passed, so be sure to pass them if they are desired.
-    :param int defer: What action should be taken on the container upon job completion?
-           FORGO (0) will leave the container untouched.
-           STOP (1) will attempt to stop the container with `docker stop` (useful for debugging).
-           RM (2) will stop the container and then forcefully remove it from the system
-           using `docker rm -f`. This is the default behavior if defer is set to None.
-    :returns: Stdout from the docker call
-    :rtype: str
-    """
-    return _docker(job, tool=tool, parameters=parameters, workDir=workDir,
-                   dockerParameters=dockerParameters, checkOutput=True, defer=defer)
-
-
-def _docker(job,
-            tool,
-            parameters=None,
-            workDir=None,
-            dockerParameters=None,
-            outfile=None,
-            checkOutput=False,
-            defer=None):
-    """
-    :param toil.Job.job job: The Job instance for the calling function.
-    :param str tool: Name of the Docker image to be used (e.g. quay.io/ucsc_cgl/samtools).
-    :param list[str] parameters: Command line arguments to be passed to the tool.
-           If list of lists: list[list[str]], then treat as successive commands chained with pipe.
-    :param str workDir: Directory to mount into the container via `-v`. Destination convention is /data
-    :param list[str] dockerParameters: Parameters to pass to Docker. Default parameters are `--rm`,
-            `--log-driver none`, and the mountpoint `-v work_dir:/data` where /data is the destination convention.
-             These defaults are removed if docker_parmaters is passed, so be sure to pass them if they are desired.
-    :param file outfile: Pipe output of Docker call to file handle
-    :param bool checkOutput: When True, this function returns docker's output.
-    :param int defer: What action should be taken on the container upon job completion?
-           FORGO (0) will leave the container untouched.
-           STOP (1) will attempt to stop the container with `docker stop` (useful for debugging).
-           RM (2) will stop the container and then forcefully remove it from the system
-           using `docker rm -f`. This is the default behavior if defer is set to None.
+    :param str tool: Name of the Docker image to be used.
+                     (e.g. quay.io/ucsc_cgl/samtools:latest)
+    :param list[str] parameters: A list of string elements.  If there are
```

Comment:
Line limit here looks less than 120 — not sure if the Toil team has a preference or not.